### PR TITLE
VACMS-5387: Add script to attach service location to health service nodes

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Relates to #5387

## Testing done
Visual / logs / console output

## QA steps
 - [x] run `lando drush scr scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php`
 - [x] after script completes (5-10 minutes), visit `admin/reports/dblog` and visually verify `Attaching service locations to health service nodes completed by drush script.` message appears

### CODE REVIEW

**Review by netboss — APPROVED** [review]
@ethanteague This looks great! Thanks!

**Comment by netboss** [comment]
@ethanteague 
QA'ed this locally and all seems to be fine. One note though, I had to run the script twice (second time after rebuilding local lando env) because the first time when I tried to access the site I got this error:

> The website encountered an unexpected error. Please try again later.
PDOException: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Temporary failure in name resolution in Drupal\Component\DependencyInjection\PhpArrayContainer->createService() (line 79 of /app/docroot/core/lib/Drupal/Component/DependencyInjection/PhpArrayContainer.php).

Not sure if this have any relation to the script, but just to double check and confirm, I rebuilt local lando env and then re-ran the script and this time all was fine.

File: scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  $nids_without_location->condition('n.type', 'health_care_local_health_service', '=');
+  $nids_without_location->condition('nfsl.entity_id', NULL, 'IS NULL');
+  $nids_without_location->fields('n', ['nid']);
+  $nids_without_location_array = $nids_without_location->execute()->fetchCol();
+
+  return $nids_without_location_array;
+}
+
+/**
+ * Creates the service location address paragraph for the service location.
```
Comment (swirtSJW) [inline scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php:39]: doc block missing param.

File: scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ */
+function update_health_health_service_nodes(&$sandbox) {
+  // Now grab all of our service location-less health service nodes.
+  $service_nodes = get_health_services_without_location();
+
+  // Set up our batch process.
+  $result_count = count($service_nodes);
+  $sandbox['total'] = $result_count;
+  $sandbox['nids_to_update'] = $service_nodes;
+  $sandbox['#finished'] = 0;
```
Comment (swirtSJW) [inline scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php:94]: It should not run these on each iteration.   It should only run this once.   All of these should be inside a  check for !isset($sandbox[total]

If the total is not set, it should run all the initial query and counts.   After that it should just update the nids_to_update,

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Unnecesary repeated queries.  See comments below.

File: scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php (hunk 2 above)
Comment (swirtSJW) [reply scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php:94]: $sandbox should keep the state.   It should not be re-declared with each run.

Comment (ethanteague) [reply scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php:94]: Doesn't work for me when I run it that way, @swirtSJW - if I put it in an isset, I end up looping over the same set of 50 nodes in each loop. 
I tried a bunch of different ways to emulate what you did in `scripts/content/VACMS-2975-update-vamc-service-nodes-2020-10.php` when I put this together (and some other ways, today), but the way the script is written now is the only way I can seem to get through all of the updates.

**Comment by swirtSJW** [comment]
Just adding some noted to track the state of the data.
This pr operates on 3,036 facility health service nodes out of a total of   12658

So approx 24% did not have defaults and they do as a result of running the script.

**Review by swirtSJW — APPROVED** [review]
Nice work Ethan.

I ran the script on tugboat
image
Completes appropriately.
image

Running it a second time results in no items to process which is good.
image


Logging is good
image

CHecked out the affected nodes.   Messaging is good.  REsulting data is good.
image

File: scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    ->log(LogLevel::INFO, 'Facility health service nodes %current nodes updated. Nodes processed: %nids', [
+      '%current' => $sandbox['current'],
+      '%nids' => implode(', ', $nids),
+    ]);
+
+  $sandbox['#finished'] = ($sandbox['current'] / $sandbox['total']);
+
+  // Log the all-finished notice.
+  if ($sandbox['#finished'] == 1) {
+    Drupal::logger('drush_scr')->log(LogLevel::INFO, 'Processing all %count health service nodes completed by drush script.', [
```
[the remark this answers is not in the document]
Comment (swirtSJW) [reply scripts/content/VACMS-5387-facility-services-add-service-location-2021-11.php:152]: Disregard

**Pull request merged**